enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/actions/AddADBAction.java

```diff
@@ -0,0 +1,144 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.cloud.oracle.actions;
+
+import com.oracle.bmc.model.BmcException;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.netbeans.modules.cloud.oracle.OCIManager;
+import org.netbeans.modules.cloud.oracle.compartment.CompartmentItem;
+import org.netbeans.modules.cloud.oracle.compartment.CompartmentNode;
+import org.netbeans.modules.cloud.oracle.database.DatabaseItem;
+import org.netbeans.modules.cloud.oracle.database.DatabaseNode;
+import org.netbeans.modules.cloud.oracle.items.OCIItem;
+import org.netbeans.modules.cloud.oracle.items.TenancyItem;
+import org.openide.DialogDescriptor;
+import org.openide.DialogDisplayer;
+import org.openide.NotifyDescriptor;
+import org.openide.NotifyDescriptor.QuickPick.Item;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Jan Horvath
+ */
+@ActionID(
+        category = "Tools",
+        id = "org.netbeans.modules.cloud.oracle.actions.AddADBAction"
+)
+@ActionRegistration( 
+        displayName = "#AddADB", 
+        asynchronous = true
+)
+
+@ActionReferences(value = {
+    @ActionReference(path = "Cloud/Oracle/Common/Actions", position = 260)
+})
+@NbBundle.Messages({
+    "AddADB=Add Oracle ADB",
+    "SelectTenancy=Select Tenancy",
+    "SelectCompartment=Select Compartment",
+    "SelectDatabase=Select Database"
+})
+public class AddADBAction implements ActionListener {
+    private static final Logger LOGGER = Logger.getLogger(AddADBAction.class.getName());
+    
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        List<TenancyItem> tenancies = OCIManager.getDefault()
+                .getConnectedProfiles()
+                .stream()
+                .map(p -> p.getTenancy())
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+        Optional<TenancyItem> selectedTenancy = chooseOneItem(tenancies, Bundle.SelectTenancy());
+        
+        Optional<CompartmentItem> selectedCompartment = Optional.empty();
+                
+        if (selectedTenancy.isPresent()) {
+            List<CompartmentItem> compartments = CompartmentNode.getCompartments().apply(selectedTenancy.get());
+            selectedCompartment = chooseOneItem(compartments, Bundle.SelectCompartment());
+        }
+        Optional<DatabaseItem> selectedDatabase = Optional.empty();
```

Comment:
why Optional<DatabaseItem> and not simply DatabaseItem that can be checked for `null` ?